api/base/serializers.py

```diff
@@ -266,6 +266,7 @@ def __init__(self, related_view=None, related_view_kwargs=None, self_view=None,
         self.related_meta = related_meta
         self.self_meta = self_meta
         self.always_embed = always_embed
+        self.source_field = source_field
```

Comment:
It was indeed because the HyperlinkedIdentityField was setting ```kwargs['source'] = *``` no matter what

I tried setting it after the ```super()``` call instead but something went wrong with that. I can try again and better document the issue, because I don't remember exactly what it was.